calculator.go

```diff
@@ -7,38 +7,40 @@ import (
 	"strings"
 )
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a float64, b ...float64) float64 {
-	for _, bb := range b {
-		a += bb
+// Add takes two (or more) numbers and returns the result of adding them together.
+func Add(a, b float64, c ...float64) float64 {
+	a += b
+	for _, cc := range c {
+		a += cc
 	}
 	return a
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
-// from the first.
-func Subtract(a float64, b ...float64) float64 {
-	for _, bb := range b {
-		a -= bb
+// Subtract takes two (or more) numbers and returns the result of subtracting them
+func Subtract(a, b float64, c ...float64) float64 {
+	a -= b
+	for _, cc := range c {
+		a -= cc
 	}
 	return a
 }
 
-// Multiply takes two numbers and returns the result of multiplying them
-func Multiply(a float64, b ...float64) float64 {
-	for _, bb := range b {
-		a *= bb
+// Multiply takes two (or more) numbers and returns the result of multiplying them
+func Multiply(a, b float64, c ...float64) float64 {
+	a *= b
+	for _, cc := range c {
+		a *= cc
 	}
 	return a
 }
 
-// Divide takes two numbers and divides the first one by the second one
-func Divide(a float64, b ...float64) (float64, error) {
-	for _, bb := range b {
-		if bb == 0 {
+// Divide takes two (or more) numbers and divides the them by each other
+func Divide(a, b float64, c ...float64) (float64, error) {
+	for _, cc := range append([]float64{b}, c...) {
```

Comment:
Seeing that the return value of `append` has no use other than for the purposes of the loop, I thought I could save an allocation  and "pass" it directly to `range`. Do you think this kind approach hurts readability at all?

Reply:
It doesn't pass the 'glanceability' test for me.

Reply:
Ok I can overwrite `c` then loop through it but I guess the next question would be if the use of `append` is even warranted here at all?
```go
c = append([]float64{b}, c...)
for _, cc := range c
...
```
The purpose is to insert `b` to the top of `c` so that it's the first element to we divide `a` by.  What I'm trying to avoid is another to write another `b==0` check and an error return. In short, is something like the following more acceptable to begin with:
```go
if b == 0 {
	return 0, errors.New("division by zero")
}
a /= b
for _, cc := range c {
	if cc == 0 {
		return 0, errors.New("division by zero")
	}
	a /= cc
}
return a, nil
```
Maybe I'm just overthinking this 😂

Reply:
In general we prefer to repeat ourselves simply, rather than be complicated!